Tasks/JenkinsQueueJob/job.ts

```diff
@@ -308,6 +320,62 @@ export class Job {
             }).auth(thisJob.queue.taskOptions.username, thisJob.queue.taskOptions.password, true);
         }
     }
+
+    downloadResults(): void {
+        var thisJob: Job = this;
+        var downloadUrl: string = Util.addUrlSegment(thisJob.executableUrl, 'team-results/zip');
+        tl.debug('downloadResults(), url:' + downloadUrl);
+
+        var downloadRequest = request.get({ url: downloadUrl, strictSSL: thisJob.queue.taskOptions.strictSSL })
+            .auth(thisJob.queue.taskOptions.username, thisJob.queue.taskOptions.password, true)
+            .on("error", err => {
+                Util.handleConnectionResetError(err); // something went bad
+                thisJob.stopWork(thisJob.queue.taskOptions.pollIntervalMillis, thisJob.state);
+            })
+            .on("response", response => {
+                tl.debug('downloadResults(), response.statusCode: ' + response.statusCode);
+                if (response.statusCode != 200) {
```

Comment:
If `response.StatusCode` is 5xx, can we warn "A Jenkins error occurred while retrieving results. Results could not be downloaded."